docs/faq/code-analysis/does-codacy-place-limits-on-the-code-analysis.md

```diff
@@ -0,0 +1,56 @@
+# Does Codacy place limits on the code analysis?
+
+Codacy uses limits when performing the analysis of your repositories to ensure good performance levels and avoid degradation of service during peak load.
+
+The following table describes these limits and includes links to more information and workarounds, if available:
+
+<table>
+<thead>
+<tr>
+<th><strong>Limit</strong></th>
+<th><strong>Value</strong></th>
+<th><strong>Rationale</strong></th>
+</tr>
+</thead>
+<tbody>
+<tr>
+<td>File size</td>
+<td>150 KB</td>
+<td>
+Large source code files are typically generated by or dependent on a third-party, and could significantly delay the analysis.<br/><br/>
+See <a href="../../troubleshooting/why-is-my-file-over-150-kb-missing/">Why is my file over 150 KB missing?</a>
+</td>
+</tr>
+<tr>
+<td>File size for coverage reports</td>
+<td>10 MB</td>
+<td>
+Codacy doesn't parse code coverage reports that are over the file size limit. See <a href="../../../coverage-reporter/troubleshooting-common-issues/#jsonparseexception-while-uploading-coverage-data">JsonParseException while uploading coverage data</a>.
+</td>
+</tr>
+<tr>
+<td>Number of files for duplication</td>
+<td>5000</td>
+<td>
+Some tools fail to calculate duplication or time out when analyzing a number of files above this number.<br/><br/>
+See <a href="../../troubleshooting/why-arent-duplication-metrics-being-calculated/">Why aren't duplication metrics being calculated?</a>
+</td>
+</tr>
+<tr>
+<td>Number of issues per file and per tool</td>
+<td>50</td>
+<td>
+Codacy limits the number of issues returned on each file by individual tools as a safeguard against degradation of performance on large or unexpected use cases.<br/><br/>
+This means that in some situations Codacy could report more issues after a push that includes fixes for the currently reported issues.
+</td>
+</tr>
+<tr>
+<td>Number of comments on the Git provider</td>
+<td>25</td>
+<td>
+Codacy limits the number of comments for reporting found issues on pull requests.
```

Comment:
Do we have a better rationale for this limit? Could it be to avoid "spamming" the pull requests with comments?

Reply:
> * Do we have a better rationale for this limit? Could it be to avoid "spamming" the pull requests with comments?

Yep. For provider other than GH Apps, each comment as a user triggers an email. Having  a ceiling on API calls to the provider is another reason.